### ISSUE TITLE
Stack overflow during sync from scratch

### ISSUE THREAD

**Issue by trapp** [ethcore/parity#1757]
```
2016-07-28 18:03:31 Syncing #1430598 10fa…439f    142 blk/s  927 tx/s  22 Mgas/s       0+35172 Qed   #1465834    0/23/25 peers     31 MiB db    2 MiB chain   47 MiB queue    5 MiB sync
2016-07-28 18:03:37 Syncing #1431176 ec22…0033    113 blk/s  466 tx/s  14 Mgas/s       0+34596 Qed   #1465834    0/24/25 peers     30 MiB db    2 MiB chain   47 MiB queue    5 MiB sync
2016-07-28 18:03:47 Syncing #1431890 d6d0…dfb8    140 blk/s  749 tx/s  25 Mgas/s       0+33892 Qed   #1465834    0/24/25 peers     32 MiB db   10 MiB chain   47 MiB queue    5 MiB sync
2016-07-28 18:03:47 Syncing #1431890 d6d0…dfb8      0 blk/s    0 tx/s   0 Mgas/s       0+33892 Qed   #1465834    0/24/25 peers     32 MiB db    7 MiB chain   47 MiB queue    5 MiB sync

thread 'IO Worker #3' has overflowed its stack
fatal runtime error: stack overflow
Aborted
```

Parity compiled on 6b1e722a6b876ce922c5f269bbd2b09933e55ffa

OS was Debian 8

**Comment by gavofyork** [comment]
was this with tracing?

might be the same root cause of #1686 or #1704, which were fixed in #1698 . could you try with the latest `master`?

**Comment by trapp** [comment]
Tracing was on, yes. I'll update to latest master, however it will probably be difficult to reproduce. Feel free to close this issue when the cause is already handled in other issues. I'll re-open it when I come across this again with the latest master.
